src/test/scala/za/co/absa/hyperdrive/trigger/persistance/RepositoryTestBase.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.persistance
+
+import java.time.LocalDateTime
+import slick.jdbc.H2Profile
+import za.co.absa.hyperdrive.trigger.models.enums.DagInstanceStatuses
+import za.co.absa.hyperdrive.trigger.models.{DagInstance, Workflow}
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+trait RepositoryTestBase extends Repository {
+  val h2Profile = H2Profile
+  override val profile = h2Profile
+  import profile.api._
+
+  def h2SchemaSetup(): Unit = {
+    val schema = DBIO.seq(
+      workflowTable.schema.create,
+      dagDefinitionTable.schema.create,
+      sensorTable.schema.create,
+      jobDefinitionTable.schema.create,
+      dagInstanceTable.schema.create,
+      jobInstanceTable.schema.create,
+      eventTable.schema.create
+    )
+    Await.result(db.run(schema), Duration.Inf)
+  }
+
+  def h2SchemaDrop(): Unit = {
+    val schema = DBIO.seq(
+      eventTable.schema.drop,
+      jobInstanceTable.schema.drop,
+      dagInstanceTable.schema.drop,
+      jobDefinitionTable.schema.drop,
+      sensorTable.schema.drop,
+      dagDefinitionTable.schema.drop,
+      workflowTable.schema.drop
+    )
+    Await.result(db.run(schema), Duration.Inf)
+  }
+
+  def clearData(): Unit = {
+    val schema = DBIO.seq(
+      eventTable.delete,
+      jobInstanceTable.delete,
+      dagInstanceTable.delete,
+      jobDefinitionTable.delete,
+      sensorTable.delete,
+      dagDefinitionTable.delete,
+      workflowTable.delete
+    )
+    Await.result(db.run(schema), Duration.Inf)
+  }
+
+  def createTestData(): Unit = {
```

Comment:
Probably `createTestData` needs to go to a separate class in the future, since not all tests require the same test data.